### DISCUSSION TITLE
REST API - Example URL and JSON combinations..

### DISCUSSION THREAD

**Discussion by nodecentral** [wger-project/wger#1232] [Show and tell]
Hi, 

While I’m very new to Wger, and to be honest I’m not entirely sure I’ve got my docker instance setup/configured correctly, I’ve been trying to learn how the REST API works.. 

First off you need to have active access codes/tokens to make a submission, and how to get those is shown in the published content - > https://wger.de/en/software/api. 

What’s not shown (at least not clearly) is how you submit (POST) information to your Wger instance…

To get the ball rolling, he’s how to log your weight..

URL - **http://IP:PORT/api/v2/weightentry/**
JSON - {"date":"2023-01-20","weight":"70"}
Curl example..
````
curl -X POST -H 'Content-Type: application/json' -H 'Authorization: Bearer  eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.iuh0b2tlbl90eXBlIjoiYWNjZXNzIiwiZXhwIjoxNjc0MjEwMTU0LCJpYXQiOjE2NzQyMDk1NTQsImp0aSI6IjkyZTE0NzY3NTJmMTRhZjc4OTk2YzNhYjkxYjAwN2FkIiwidXNlcl9pZCI6Mn0.jPhD8EfuPOUkA-b0JTw7O8DSgl5nNHEAKWi023zNyLc' -d '{"date":"2023-01-20","weight":"70"}' http://192.168.102.134:8008/api/v2/weightentry/
````
I’ll add more when I’ve worked some of the others out… :-) 

If anyone reading this has working examples, please share them below..

**Comment by mynamewastaken** [comment]
No reverse engineering needed, you can use swagger to browse and hit the various endpoints - https://wger.de/api/v2/schema/ui#/weightentry/weightentry_create
